### PR TITLE
config: move config update interface from pd to status server

### PR DESCRIPTION
**Pull request by NingLin-P**
### What is changed and how it works?

What's Changed:
move config update interface from pd to status server

### Check List 

Tests 

- Unit test
- Integration test

### CODE REVIEW

File: components/configuration/configuration_derive/src/lib.rs [hunk 1]
```diff
@@ -230,6 +232,43 @@ fn diff(fields: &Punctuated<Field, Comma>, creat_name: &Ident) -> Result<TokenSt
     })
 }
 
+fn typed(fields: &Punctuated<Field, Comma>, creat_name: &Ident) -> Result<TokenStream> {
```
Comment (overvenus) [inline components/configuration/configuration_derive/src/lib.rs:235]: Could you add some comments about the `typed`?

Comment (NingLin-P) [reply components/configuration/configuration_derive/src/lib.rs:235]: I had added comments on the `Configuration` trait, PTAL.

File: components/pd_client/src/util.rs [hunk 2]
```diff
@@ -28,41 +26,14 @@ use tikv_util::security::SecurityManager;
...
-
 pub struct Inner {
     env: Arc<Environment>,
     pub hb_sender: Either<
         Option<ClientDuplexSender<RegionHeartbeatRequest>>,
         UnboundedSender<RegionHeartbeatRequest>,
     >,
     pub hb_receiver: Either<Option<ClientDuplexReceiver<RegionHeartbeatResponse>>, Task>,
-    pub client_stub: PdClientStub,
+    pub client_stub: PdClient,
```
Comment (rleungx) [inline components/pd_client/src/util.rs:36]: #5594 renames the `PdClient` to `PdClientStub`, should we keep the name?

Comment (NingLin-P) [reply components/pd_client/src/util.rs:36]: Sorry I miss that, I will change it back.

**Pull request merged**